src/verify-app-guide/verification-steps/application-dependencies.component.js

```diff
@@ -0,0 +1,142 @@
+import React, { useState, useEffect } from "react";
+import CodeOutput from "../../shared/code-output.component";
+
+export default React.forwardRef(function ApplicationDependencies(
+  { app, stepNumber },
+  ref
+) {
+  const [isExpanded, setIsExpanded] = useState(false);
+  const [error, setError] = useState(null);
+  const [sharedDependencies, setSharedDependencies] = useState([]);
+  const [missingDeps, setMissingDeps] = useState([]);
+
+  if (ref) {
+    ref.current = {
+      runTest,
+      resetError() {
+        setError(false);
+      },
+    };
+  }
+
+  useEffect(() => {
+    if (missingDeps.length > 0) {
+      setError(true);
+      setIsExpanded(true);
+    }
+  }, [missingDeps]);
+
+  return (
+    <div className="step">
+      <div
+        role="button"
+        tabIndex={0}
+        className="step-header"
+        onClick={toggleExpanded}
+      >
+        <div className="step-number">{stepNumber}</div>
+        <div className="step-summary">
+          Checking if application has shared dependencies
+        </div>
+      </div>
+      {isExpanded && (
+        <div className="step-detail">
+          {sharedDependencies && (
+            <>
+              <div>Your app has the following shared dependencies:</div>
+              <CodeOutput code={sharedDependencies} />
+            </>
+          )}
+          {error && missingDeps.length > 0 ? (
+            <>
+              <div>Found missing dependencies:</div>
+              <CodeOutput code={missingDeps} />
+              <div>
+                Add them to the playground (using import-map-overrides on
+                bottom-right corner) to continue, otherwise, an error will be
+                thrown in the next step!
+              </div>
+            </>
+          ) : null}
+        </div>
+      )}
+    </div>
+  );
+
+  function toggleExpanded() {
+    setIsExpanded(!isExpanded);
+  }
+
+  function runTest() {
+    return getAppDependencies({ name: app.name })
+      .then((appSharedDeps) => {
+        setSharedDependencies(appSharedDeps);
+        const missingDeps = getMissingDeps(appSharedDeps);
+        if (missingDeps.length > 0) {
+          setMissingDeps(missingDeps);
+          throw new Error("Missing shared dependencies");
+        }
+
+        toggleExpanded();
+      })
+      .catch((err) => {
+        setError(err);
+        console.error(err);
+        throw err;
+      });
+  }
+});
+
+export const getAppDependencies = ({ name, url }) => {
```

Comment:
Wow this is quite fancy. It works! Impressive

Reply:
It sure seem like a hack to me... I wondered if there was some API to get those dependencies (like, System.import() something, and ask System what dependencies are needed for a given specifier, but I found nothing in the API... so just to unblock me, I searched for the deps in the string haha
I learned about greed/lazy regex with you, so I could capture what's inside `[` until the first `]` 😁

Reply:
SystemJS does not allow for introspection of a module's dependencies. See the following links:

1. Dependencies are stored in the "load" object:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L184
2. The load object is stored in the SystemJS registry:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L167
3. The registry is a private Symbol, so it can't be accessed by code outside of SystemJS:
https://github.com/systemjs/systemjs/blob/c977894618c83928c0b843bf998f97bf2e0ffb1c/src/system-core.js#L22

Reply:
I see... thanks!